docs/resolve/relationships.rst

```diff
@@ -142,50 +142,98 @@ This is by design because you wouldn't want one component to dispose the ``B`` o
+
+Another corollary of this behaving like calling ``Resolve<B>()`` is that **Lifetime scopes are respected** using this relationship type. If you register an object as ``InstancePerDependency()`` and call the ``Func<B>`` multiple times, you'll get a new instance each time. However, if you register an object as ``SingleInstance()`` and call the ``Func<B>`` to resolve the object more than once, you will get *the same object instance every time*.
 
 An example of this relationship looks like:
 
 .. sourcecode:: csharp
+    public class B
+    {
+      public B() {}
+      
+      public void DoSomething() {}
+    }
 
     public class A
     {
-      Func<B> _b;
+      Func<B> _newB;
 
-      public A(Func<B> b) { _b = b; }
+      public A(Func<B> b) { _newB = b; }
 
       public void M()
       {
-          var b = _b();
+          var b = _newB();
           b.DoSomething();
       }
     }
 
 
 Parameterized Instantiation (Func<X, Y, B>)
 -------------------------------------------
-You can also use an *auto-generated factory* to pass strongly-typed parameters to the resolution function. This is an alternative to :doc:`passing parameters during registration <../register/parameters>` or :doc:`passing during manual resolution <../resolve/parameters>`:
+You can also use an *auto-generated factory* to provide parameters when creating an new instance of the object, where the constructor of the object calls for some additional parameters. As in the case of Dynamic Instantiation above, we're still doing something equivalent to calling ``.Resolve<B>()``, but in this case we're passing strongly-typed parameters to the resolution function. This is an alternative to :doc:`passing parameters during registration <../register/parameters>` or :doc:`passing during manual resolution <../resolve/parameters>`:
 
 .. sourcecode:: csharp
+    public class B
+    {
+      public B(string someString, int id) {}
+      
+      public void DoSomething() {}
+    }
 
     public class A
     {
-        Func<int, string, B> _b;
+        Func<int, string, B> _newB;
 
-        public A(Func<int, string, B> b) { _b = b }
+        public A(Func<int, string, B> b) { _newB = b }
 
         public void M()
         {
-            var b = _b(42, "http://hel.owr.ld");
+            var b = _newB(42, "http://hell.owor.ld");
+            b.DoSomething();
+        }
+    }
+
+Note that since we're ``.Resolve()``ing the instantiation, rather than actually directly calling the constructor we don't need to declare the parameters in the same order as they appear in the ctor, nor do we necessarily need to provide *all* the parameters listed in that constructor. If some of the constructor's parameters could have been resolved by the Autofac Container already, then those parameters can be omitted from the ``Func`` signature being declared, so you only need to list the types that the Container can't resolve.
```

Comment:
Inline code attached to other things doesn't really render well. Trying to unify references to the function type so it's consistent across all of them (`Func<B>` or `Func<X, Y, B>` rather than also having `Func`). Using "lifetime scope" instead of "container" to drive home that lifetime scope is obeyed. (Registrations can be added to the lifetime scope that don't exist in the container.)

```suggestion
Note that since we're resolving the instance rather than directly calling the constructor we don't need to declare the parameters in the same order as they appear in the constructor definition, nor do we need to provide *all* the parameters listed in the constructor. If some of the constructor's parameters can be resolved by the lifetime scope those parameters can be omitted from the ``Func<X, Y, B>`` signature being declared. You only need to list the types that the scope can't resolve.
```

Reply:
Note `lifetime scope those parameters` still needs the `, then` in the middle to avoid mis-parsing it.

```suggestion
Note that since we're resolving the instance rather than directly calling the constructor we don't need to declare the parameters in the same order as they appear in the constructor definition, nor do we need to provide *all* the parameters listed in the constructor. If some of the constructor's parameters can be resolved by the lifetime scope, then those parameters can be omitted from the ``Func<X, Y, B>`` signature being declared. You only *need* to list the types that the scope can't resolve.
```

Reply:
Yep. Looks good.